components/post_markdown/post_markdown.jsx

```diff
@@ -63,17 +63,25 @@ export default class PostMarkdown extends React.PureComponent {
         const channelNamesMap = this.props.post && this.props.post.props && this.props.post.props.channel_mentions;
 
         let {message} = this.props;
-        const {post} = this.props;
-        const options = {...this.props.options};
+        const {post, allMentionKeys, mentionKeysWithoutGroups, options} = this.props;
 
         this.props.pluginHooks.forEach((o) => {
             if (o && o.hook && post) {
                 message = o.hook(post, message);
             }
         });
 
+        let mentionKeys = [];
         if (post && post.props) {
-            options.mentionHighlight = !post.props.mentionHighlightDisabled;
+            mentionKeys = allMentionKeys;
+
+            if (post.props.disable_group_highlight) {
+                mentionKeys = mentionKeysWithoutGroups;
+            }
+
+            if (post.props.disable_channel_highlight) {
```

Comment:
if you're using a different prop now you should replace all instances of the previous prop with this - or remove the old logic added here https://github.com/mattermost/mattermost-webapp/pull/4916/

Reply:
I'd say we should stick with the old prop names. As bad as it feels to mix up camel case and snake case here, being consistently bad with it is preferable.